services/user.js

```diff
@@ -379,34 +384,38 @@ class User {
         }
         updatedUser.studies = params.studies;
 
-        if (params?.status){
-            if (! [USER.STATUSES.ACTIVE, USER.STATUSES.INACTIVE].includes(params.status))
-                throw new Error(ERROR.INVALID_USER_STATUS);
-
-            updatedUser.status = params.status
-        }
-
         // Check if an organization is required and missing for the user's role
         const userHasOrg = Boolean(user[0]?.organization?.orgID);
         if (!userHasOrg && [USER.ROLES.DC_POC, USER.ROLES.ORG_OWNER, USER.ROLES.SUBMITTER, USER.ROLES.FEDERAL_MONITOR].includes(updatedUser.role || user[0]?.role)) {
             throw new Error(ERROR.USER_ORG_REQUIRED);
         }
+        return await this.updateUserInfo(user[0], updatedUser, params.userID, params.status, params.role, params?.studies);
+    }
+
+    async updateUserInfo(prevUser, updatedUser, userID, status, role, approvedStudyIDs) {
+        // add studies to user.
+        const validStudies = await this.#findApprovedStudies(approvedStudyIDs);
+        if (validStudies?.length !== approvedStudyIDs?.length) {
+            throw new Error(ERROR.INVALID_NOT_APPROVED_STUDIES);
+        }
+
+        if (validStudies && approvedStudyIDs) {
+            updatedUser.studies = validStudies; // return approved studies dynamically with all properties of studies
```

Comment:
Hi Young, can't set updatedUser.studies = validStudies; before save updated user.   Only save studyIDs but not other fields.   When call get user, the approvedStudies for the user will be retried dynamically.

Reply:
I've applied some change. Could you double check? **Only save studyIDs but not other fields.** I've just relocated the updatedUser.studies = validStudies; Thanks.